app/src/main/java/com/github/polybooks/database/FBInterestDatabase.kt

```diff
@@ -179,20 +212,116 @@ class FBInterestDatabase: InterestDatabase {
     }
 
     /**
-     * Get the interests of the specified user
+     * Get the interests of the current user
+     * */
+    override fun getCurrentUserInterests(): CompletableFuture<List<Interest>> {
+        val user = firebaseUserToUser(FirebaseProvider.getAuth().currentUser)
+        return if(user is LoggedUser) {
+            getLoggedUserInterests(user)
+        } else {
+            getLocalUserInterests()
+        }
+    }
+
+    /**
+     * Sets the interests of the current user.
+     * @return A Future to receive confirmation of success/failure asynchronously
+     * */
+    override fun setCurrentUserInterests(interests: List<Interest>): CompletableFuture<List<Interest>> {
+        val user = firebaseUserToUser(FirebaseProvider.getAuth().currentUser)
+        return if(user is LoggedUser) {
+            setLoggedUserInterests(user, interests)
+        } else {
+            setLocalUserInterests(interests)
+        }
+    }
+
+    /**
+     * Get the interests of the specified logged user
      * TODO: Might need to add an authentication token to restrict authenticated users to only modify their interests.
      * */
-    override fun getUserInterests(user: User): CompletableFuture<Triple<List<Field>, List<Semester>, List<Course>>> {
-        TODO("Not yet implemented")
+    fun getLoggedUserInterests(user: LoggedUser): CompletableFuture<List<Interest>> {
```

Comment:
Shouldn't this be private? There are several methods that could be set private I think.

Reply:
Look, I'll be honest. It was private, and it should be private.
But I couldn't manage to mock "being a logged user" in the firebase emulator, so I switched it to public so that I can test the function directly instead of calling its "mother function"

Reply:
What we do is register a new user and then delete is at the end. Smarter would be to just log in as a user and disconnect at the end